### DISCUSSION TITLE
immersion controller as a switch

### DISCUSSION THREAD

**Discussion by balajibalasa** [WillCodeForCats/solaredge-modbus-multi#19] [Ideas]
Hi, can anyone let me know is there a way to get the immersion controller paired with the solar edge inverter can be made available as a switch in HA.  Posting pairing it is available as a device in the APP on iOS..

**Comment by WillCodeForCats** [comment]
What's an "immersion controller" and what does it have to do with solaredge?

**Reply by balajibalasa** [reply]
Hi, Solaredge makes an immersion controller to divert the excess solar generated energy to the grid instead to the hot water cylinder, where the water is heated through the immersion heater.  The immersion heater is controlled by the SolarEdge inverter through ZIGBEE and is visible as a device in the App and the web portal , if you need more information I have pasted the url here. 

https://www.solaredge.com/sites/default/files/se-device-control-immersion-heater-controller-installation-guide.pdf 

Apart from the immersion heater , there are other SolarEdge devices which can be paired directly to the inverter for smartly diverting the electricity increasing the self use from the system.

If a switch can be provided in the form of entity in HA we can use the same through a range of automations and scenes.

With rising energy costs , rather than exporting unused energy , this little clever device diverts the electricity to the immersion heater.  

Thanks in advance.

**Comment by WillCodeForCats** [comment]
I'm in the US so that probably explains why I've never heard of that device.

This integration uses the documented Sunspec modbus protocol (NOT the SolarEdge API which is completely different), so basically the answer is if it's part of Sunspec then yes but if it's not in Sunspec then no.

**Reply by balajibalasa** [reply]
Thank you for your response here. I will reach out to Tech support here locally and ask some more questions , if I am successful is getting hold of any tech documents , will that be helpful for you to update the code base. Let me know..
